index.ts

```diff
@@ -1,139 +1,199 @@
-import { join } from 'path';
+import path from 'path';
 import { Plugin } from 'rollup';
 import {
-	CompilerOptions,
-	findConfigFile,
-	nodeModuleNameResolver,
-	parseConfigFileTextToJson,
-	sys,
+  CompilerOptions,
+  findConfigFile,
+  nodeModuleNameResolver,
+  parseConfigFileTextToJson,
+  sys,
 } from 'typescript';
 
 export const typescriptPaths = ({
-	absolute = true,
-	nonRelative = false,
-	preserveExtensions = false,
-	tsConfigPath = findConfigFile('./', sys.fileExists),
-	transform,
+  absolute = true,
+  nonRelative = false,
+  preserveExtensions = false,
+  tsConfigPath = findConfigFile('./', sys.fileExists),
+  transform,
 }: Options = {}): Plugin => {
-	const { compilerOptions, outDir } = getTsConfig(tsConfigPath);
-
-	return {
-		name: 'resolve-typescript-paths',
-		resolveId: (importee: string, importer?: string) => {
-			const enabled = Boolean(
-				compilerOptions.paths || (compilerOptions.baseUrl && nonRelative),
-			);
-
-			if (
-				typeof importer === 'undefined' ||
-				importee.startsWith('\0') ||
-				!enabled
-			) {
-				return null;
-			}
-
-			const hasMatchingPath =
-				!!compilerOptions.paths &&
-				Object.keys(compilerOptions.paths).some((path) =>
-					new RegExp('^' + path.replace('*', '.+') + '$').test(importee),
-				);
-
-			if (!hasMatchingPath && !nonRelative) {
-				return null;
-			}
-
-			if (importee.startsWith('.')) {
-				return null; // never resolve relative modules, only non-relative
-			}
-
-			const { resolvedModule } = nodeModuleNameResolver(
-				importee,
-				importer,
-				compilerOptions,
-				sys,
-			);
-
-			if (!resolvedModule) {
-				return null;
-			}
-
-			const { resolvedFileName } = resolvedModule;
-
-			if (!resolvedFileName || resolvedFileName.endsWith('.d.ts')) {
-				return null;
-			}
-
-			const targetFileName = join(
-				outDir,
-				preserveExtensions
-					? resolvedFileName
-					: resolvedFileName.replace(/\.tsx?$/i, '.js'),
-			);
-
-			const resolved = absolute
-				? sys.resolvePath(targetFileName)
-				: targetFileName;
-
-			return transform ? transform(resolved) : resolved;
-		},
-	};
+  const { compilerOptions } = getTsConfig(tsConfigPath);
+  const outDir = compilerOptions.outDir ?? '.';
```

Comment:
This was pointed out here already: https://github.com/simonhaenisch/rollup-plugin-typescript-paths/issues/14#issuecomment-1445171583

Reply:
Right in my case, `outDir` was always the default (**"."**) since I've defined it in my `compilerOptions` -> never was read in.

Since the `resolvedFileName` is already an absolute path, joining (with `path.join()`) it with the default `outDir`  (**"."**) resulted in the correct absolute path, as `resolvedFileName` seems already correct?

Example `resolvedFileName`: **"C:/path/to/my/monorepo/packages/dtif-to-svg/src/d3.ts"**

However, now that it reads in the `outDir` correctly due to your suggested `getParsedCommandLineOfConfigFile()` method that works like a charm, the `outDir` is something like **"C:/path/to/my/monorepo/packages/dtif-to-svg/dist"** which I defined in the `compilerOptions` of my `tsconfig.json`. However, now `path.join()` will join these both absolute paths that result in: **"C:/path/to/my/monorepo/packages\dtif-to-svg\dist\C:/path/to/my/monorepo/packages\dtif-to-svg\src\d3.ts"**. And thats not correct.

-> For me it works best without any defined `outDir` as `resolvedFileName` already points to the correct location
-> Do we actually need to define the `outDir`?

Reply:
In the same issue comments I think further down there's a solution with using `path.relative` instead of `path.join`.